### PR TITLE
Auto Detect Tokens (GUI)

### PR DESCRIPTION
**Pull request by Jon-edge**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

https://github.com/EdgeApp/edge-currency-accountbased/pull/678
https://github.com/EdgeApp/edge-core-js/pull/572

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [x] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205654034220620

### CODE REVIEW

File: src/reducers/CoreReducer.ts [hunk 1]
```diff
@@ -52,5 +54,32 @@ export const core = combineReducers<CoreState, Action>({
...
     }
+  },
+
+  enabledDetectedTokens(state: { [walletId: string]: string[] } = {}, action: Action): { [walletId: string]: string[] } {
+    switch (action.type) {
+      case 'CORE/NEW_TOKENS': {
+        const newState = { ...state }
+        const { walletId, enablingTokenIds } = action.data
+
+        if (enablingTokenIds.length === 0) showError(`enabledDetectedTokens: Received empty tokens array for ${walletId}`)
```
Comment (swansontec) [inline src/reducers/CoreReducer.ts:65]: I don't see why this test is needed. Was it for debugging purposes?

File: src/__tests__/scenes/__snapshots__/CryptoExchangeQuoteScene.test.tsx.snap [hunk 2]
```diff
@@ -1393,5 +1393,17 @@ exports[`CryptoExchangeQuoteScreenComponent should render with loading props 1`]
       </View>
     </RCTScrollView>
   </View>
+  <View
```
Comment (swansontec) [inline src/__tests__/scenes/__snapshots__/CryptoExchangeQuoteScene.test.tsx.snap:1396]: Don't these test changes go with the previous commit? This one didn't change any JSX.

**Review by swansontec — APPROVED** [review]
I am surprised by this PR. I thought we were going to use the new-transaction drop-down,  not the bottom notification area. The bottom area is more for persistent notifications that sit around a long time, not for one-shot events like this.

However, I do see that it works, so I presume you have a good reason for this.

Besides this, I would like you to delete the `showError` in the reducer - reducers shouldn't have side-effects like this.

File: src/reducers/CoreReducer.ts (hunk 1 above)
Comment (Jon-edge) [reply src/reducers/CoreReducer.ts:65]: No, it's an assertion

**Comment by Jon-edge** [comment]
> I am surprised by this PR. I thought we were going to use the new-transaction drop-down, not the bottom notification area. The bottom area is more for persistent notifications that sit around a long time, not for one-shot events like this.
> 
> However, I do see that it works, so I presume you have a good reason for this.
> 
> Besides this, I would like you to delete the `showError` in the reducer - reducers shouldn't have side-effects like this.

Fair point, we should probably add some persistence duration options later. Honestly there wasn't any spec on the appearance of the notification beyond the term "notification" so I took some liberties. 

The problem with the airship was that receiving a new token would overlap the auto-detect message with the tx received airship making it kind of messy. I also wanted to just get this all laid out so we have the option to eventually consolidate notifications neatly into a nicer looking component that can handle displaying multiple items. We could even expand the notification feature later with a dedicated scene and settings for the user to enable/disable notifications of interest.

File: src/__tests__/scenes/__snapshots__/CryptoExchangeQuoteScene.test.tsx.snap (hunk 2 above)
Comment (Jon-edge) [reply src/__tests__/scenes/__snapshots__/CryptoExchangeQuoteScene.test.tsx.snap:1396]: I went back and updated each commit with `yarn test -u`

**Pull request merged**